src/test/ui/issues/issue-35668.stderr

```diff
@@ -6,10 +6,10 @@ LL |     a.iter().map(|a| a*a)
    |                      |
    |                      &T
    |
-help: consider restricting type parameter `T`
+help: consider introducing a `where` bound, but there might be an alternative better way to express this requirement
    |
-LL | fn func<'a, T: std::ops::Mul<Output = &T>>(a: &'a [T]) -> impl Iterator<Item=&'a T> {
-   |              ++++++++++++++++++++++++++++
+LL | fn func<'a, T>(a: &'a [T]) -> impl Iterator<Item=&'a T> where &T: Mul<&T> {
+   |                                                         +++++++++++++++++
```

Comment:
Is this suggestion change correct?

Reply:
Yes, albeit underconstrained on `Output`. It actually fixes the previously incorrect suggestion of `T: std::ops::Mul<Output = f64>`. In that case, `Rhs = Self` means only `T * T` is valid, whereas in this case you specifically need `T * f64` here.